src/vs/server/node/remoteAgentEnvironmentImpl.ts

```diff
@@ -308,7 +311,7 @@ export class RemoteAgentEnvironmentChannel implements IServerChannel {
 	}
 
 	private async _scanExtensions(language: string, extensionDevelopmentPath?: string[]): Promise<IExtensionDescription[]> {
-		// Ensure that the language packs are available
+		await this._ensureLanguagePackIsInstalled(language);
```

Comment:
I wonder if this impacts performance of starting remote extension host. I am assuming `_ensureLanguagePackIsInstalled` is executed even if the language pack extension is installed. Is it possible to know upfront if the language pack for the given language is available?

Reply:
This PR does that by reading the NlsConfig and seeing if one is already installed

Reply:
I see. If I am not wrong in a normal case reading the NlsConfig is done twice, once for checking if extension exists and second when we translate extensions. Is not this redundant and can we reduce this to once?

Reply:
We actually run that line:
https://github.com/microsoft/vscode/blob/ff74790f8040aab42e1331474d33d6bfbc915dc3/src/vs/server/node/extensionsScannerService.ts#L41
_per extension_. I don't know the history of that decision but maybe that should be cleaned up since `getNLSConfiguration` reads in a file every time:
https://github.com/microsoft/vscode/blob/ff74790f8040aab42e1331474d33d6bfbc915dc3/src/vs/server/node/remoteLanguagePacks.ts#L21-L37

Reply:
I don't know too much about this code. Just going off of what I'm seeing.

Reply:
@sandy081 considering we run that line a lot, but running it once more shouldn't impact performance that much and would rather defer optimizing `getNLSConfiguration` for another PR

Reply:
On second thought, I decided to use the `ILanguagePackService` which caches what language packs are installed (and keeps an updated list). Considering I am trying to know if the language pack is installed, it felt like the perfect pack for the job.